docs/proposals/project-user-resource-quotas.md

```diff
@@ -0,0 +1,221 @@
+| DigitalOcean | Y                         |                                                        |                                  |
+| GCP          | Y                         |                                                        |                                  |
+| Hetzner      | Y                         |                                                        |                                  |
+| Openstack    | Y                         |                                                        |                                  |
+| KubeVirt     | N                         | We set the requested size directly into MachineDeployment |                                  |
+| Nutanix      | TBD                       | TBD                                                    |                                  |
+| Equinox      | Y                         |                                                        |                                  |
+| vSphere      | N                         | We set the requested size directly into MachineDeployment |                                  |
+
+
+### Enforcing Quotas - Implementation variants
+
+I see 2 possible ways to implement the enforcing of project quotas through MachineDeployments. 
+
+One is through a MachineDeployment webhook, although it requires moving the MD's to the seed cluster, because the webhook 
+needs to be out of reach for the user-cluster users. The pros of that is that users can get directly notified if they 
+can't create a MachineDeployment, but the con is that we would need to do a migration and change the shared code with KubeOne.
+
+In the other approach, the machine-controller would be responsible for checking the quota before creating nodes for a MachineDeployment.
+The pro is that it wouldn't be a big change, the con is that it won't be so visible to users, we could get questions "why are my nodes not being created".
+We can show the info that there is not enough resource quota left to create the nodes, but users would only see that in MachineDeployment
+Events. So it would be something like Pods not getting scheduled because there is no space.
+
+
+#### 1. Webhook for MachineDeployments
+
+This variant is dependent on migrating the MachineDeployments from being user-cluster resources to being seed-cluster resources.
+
+If the MD's are on the seed cluster, we can control their creation/modification through a validation webhook which would check with the 
+project's resource quota if the MD fits. If not, we would just deny the request.
+
+The problem here is if its possible and at what cost to migrate the MD's, while thinking about KubeOne as well.
+We could eventually dynamically set the kubeconfig of the cluster where the machine-controller watches for MachineDeployments, 
+and set it to the seed-cluster in case of KKP. But then, as MachineDeployments are cluster-scoped we will need additional 
+cluster info label or something on them.
```

Comment:
> But then, as MachineDeployments are cluster-scoped we will need additional 
cluster info label or something on them.

Not sure if this is what you mean, but `MachineDeployments` are namespaced-scoped (they just all sit in `kube-system` on the user cluster), so they _could_ run in the cluster namespace on the seed and that maps them to the right cluster. Still, quite a big change, but just wanted to add this information.

Reply:
Right, removed. Somehow I had it programmed in my head that its cluster-scoped